### PR TITLE
Make each RS its own schema with a common Metadata data group

### PR DESCRIPTION
**Pull request by nealkruis**
This PR moves to a new structure where every RS has the metadata in it (instead of the rs_instance being a data element next to the metadata).

### CODE REVIEW

File: schema205/json_translate.py [hunk 1]
```diff
@@ -316,10 +318,9 @@ def _load_meta_info(self, schema_section):
         if 'Root Data Group' in schema_section:
             self._schema['$ref'] = self._schema_name + '.schema.json#/definitions/' + schema_section['Root Data Group']
         # Create a dictionary of available external objects for reference
-        refs = list()
+        refs = [self._schema_name]
```
Comment (nealkruis) [inline schema205/json_translate.py:321]: @tanaya-bigladder this change surprised me. The translator was choosing references before the local file. I'm not sure how it was finding the right data groups before...

File: schema205/json_translate.py [hunk 2]
```diff
@@ -221,6 +221,8 @@ def _get_simple_constraints(self, constraints_str, target_dict):
                         target_dict[mx] = maximum
                     elif '%' in c:
                         target_dict['multipleOf'] = int(numerical_value)
+                    elif 'string' in target_dict['type']:
```
Comment (nealkruis) [inline schema205/json_translate.py:224]: @tanaya-bigladder also note the addition of the string pattern constraint. I needed to add this to validate that the `data_model` in all the examples was `"ASHRAE 205"`. Maybe there's a better way to handle this?

File: schema205/json_translate.py (hunk 1 above)
Comment (tanaya-bigladder) [reply schema205/json_translate.py:321]: This isn't the code I'm working with on my branch; maybe one of my changes hadn't been merged properly? If this works, it's fine with me, otherwise on my branch I have the order reversed: 
`        refs = list()
        if 'References' in schema_section:
            refs = schema_section['References']
        refs.append(self._schema_name)
`

**Review by tanaya-bigladder — APPROVED** [review]
Yup!

**Review by TimMcDowell — COMMENTED** [review]
This all seems to work fine.  I really don't like the produced JSON schema using references for everything.  I know this works, but is very hard to follow when just reading the schema.  Since this is a normative part of the standard we may want to revisit how the schema is produced.
